source/ed/ed_reader.cpp

```diff
@@ -619,6 +622,135 @@ void EdReader::fill_stop_points(nt::Data& data, pqxx::work& work) {
+        }
+    } else {
+        LOG4CPLUS_ERROR(log, "pathway.to_stop_id not match with a stop point uri " << sp_id);
+    }
+}
+
+void EdReader::fill_access_points(nt::Data& data, pqxx::work& work) {
+    // access_point
+    std::string request =
+        "SELECT id, name, uri, "
+        "ST_X(coord::geometry) as lon, ST_Y(coord::geometry) as lat, "
+        "stop_code, "
+        "parent_station "
+        "FROM navitia.access_point";
+
+    pqxx::result result = work.exec(request);
+    for (auto const_it = result.begin(); const_it != result.end(); ++const_it) {
+        auto* ap = new nt::AccessPoint();
+        const_it["uri"].to(ap->uri);
+        const_it["name"].to(ap->name);
+        const_it["stop_code"].to(ap->stop_code);
+        ap->coord.set_lon(const_it["lon"].as<double>());
+        ap->coord.set_lat(const_it["lat"].as<double>());
+        // parent station is Stop Area
+        auto sa_key = uri_to_idx_stop_area.find("stop_area:" + const_it["parent_station"].as<std::string>());
+        if (sa_key != uri_to_idx_stop_area.end()) {
+            auto sa = stop_area_map.find(sa_key->second);
+            if (sa != stop_area_map.end()) {
+                ap->parent_station = sa->second;
+            }
+        } else {
+            LOG4CPLUS_ERROR(log, "access_point.parent_station not match with a stop area uri "
+                                     << const_it["parent_station"].as<std::string>());
+        }
+
+        // store access_point temporarily before finishing in a SP link
+        access_point_map[const_it["uri"].as<std::string>()] = ap;
+    }
+
+    // Pathway
+    request =
+        "SELECT id, name, uri, "
+        "from_stop_id, "
+        "to_stop_id, "
+        "pathway_mode, "
+        "is_bidirectional, "
+        "length, "
+        "traversal_time, "
+        "stair_count, "
+        "max_slope, "
+        "min_width, "
+        "signposted_as, "
+        "reversed_signposted_as "
+        "FROM navitia.pathway";
+
+    result = work.exec(request);
+    for (auto const_it = result.begin(); const_it != result.end(); ++const_it) {
+        std::string from_stop_id;
+        const_it["from_stop_id"].to(from_stop_id);
+        std::string to_stop_id;
+        const_it["to_stop_id"].to(to_stop_id);
+
+        // Access Point URI match for from_stop_id
+        // so, to_stop_id have to be a StopPoint
+        auto from_access_p = access_point_map.find(from_stop_id);
+        if (from_access_p != access_point_map.end()) {
+            fill_access_point_field(from_access_p->second, const_it, true, to_stop_id);
+            continue;
+        }
+        // Access Point URI match for to_stop_id
+        // so, from_stop_id have to be a StopPoint
+        auto to_access_p = access_point_map.find(to_stop_id);
+        if (to_access_p != access_point_map.end()) {
+            fill_access_point_field(to_access_p->second, const_it, false, from_stop_id);
+            continue;
+        }
     }
```

Comment:
Maybe log an error here, in case neither from_stop_id and to_stop_id are Access Point URI

Reply:
this case can exist with a SP->SP connection but we don't want to handle for the moment.
I'll add a comment to explain the unhandled case